vdb/lib/config.py

```diff
@@ -191,3 +191,6 @@
     "ubuntu": ["ubuntu"],
     "wolfi": ["wolfi"]
 }
+
+# Used for search_by_any, ordered by most common to least common
+ADVISORY_PREFIXES = ["CVE-", "ALSA-", "MAL-", "GHSA-", "DSA-", "DLA-", "ALBA-", "RLSA-","ALEA-", "RUSTSEC-", "DTSA-", "GO-", "PSF-", "PYSEC-", "OSV-", "RXSA-", "RSEC-", "UVI-"]
```

Comment:
NPM- needs to be included.

Reply:
@prabhu These are the only prefixes which exist in the latest prebuilt database... if NPM should be there, we might need to look into why there are no entries for it (I did `SELECT substr(cve_index.cve_id, 0, 5), count(substr(cve_index.cve_id, 0, 5)) FROM cve_index GROUP BY substr(cve_index.cve_id, 0, 5) ORDER BY count(substr(cve_index.cve_id, 0, 5)) DESC` to identify).

Reply:
I think NPM- comes from the live audit in depscan directly. We must also be careful in that our logic currently always [prefers](https://github.com/AppThreat/vulnerability-db/blob/e0e3ec87efa18fb63c57a57c6020301add99fb0f/vdb/lib/aqua.py#L273) [CVE](https://github.com/AppThreat/vulnerability-db/blob/e0e3ec87efa18fb63c57a57c6020301add99fb0f/vdb/lib/aqua.py#L842) [ids](https://github.com/AppThreat/vulnerability-db/blob/e0e3ec87efa18fb63c57a57c6020301add99fb0f/vdb/lib/aqua.py#L852) in the feeds and falls back to other id only if there are no CVEs. So `CVE-` is the only prefix we can support for most data. I think I am ok with a fixed set of prefixes.

Reply:
@prabhu This is a fixed set of prefixes - perhaps I am not understanding something. Checking for more than the three prefixes currently being checked (CVE, GHSA, MAL) is about as non-resource-intensive an operation as I could hope for, so why restrict the logic? If it's going through the search_by_any function, one of the search functions is going to be run regardless, so it seems like we might as well make it the right one.

Reply:
Yes. We both are talking about different topics. Will explain over the call sometime. We can include the fixed set of prefixes. We may have to include some more since we currently exclude multiple feeds such as oss fuzz, android, linux etc in the vdb. https://github.com/AppThreat/vulnerability-db/blob/e0e3ec87efa18fb63c57a57c6020301add99fb0f/vdb/lib/config.py#L84

Reply:
@prabhu I took your suggestion to use a regex.